low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -155,6 +156,42 @@ def get_platform_sensors_info(self):
         sensor_data = self.build_encl_platform_sensors_data(sensor_list)
         return sensor_data
 
+    @staticmethod
+    def get_fan_specfics(fan):
+        return {
+            "uid": fan.get("durable-id", "NA"),
+            "location": fan.get("location", "NA"),
+            "status": fan.get("status", "NA"),
+            "speed": fan.get("speed", "NA"),
+            "serial-number": fan.get("serial-number", "N/A"),
+            "part-number": fan.get("part-number", "N/A"),
+            "health": fan.get("health", "OK"),
+        }
+
+    def get_fanmodules_info(self):
+        response = []
+        fanmoduels_data = self.get_realstor_encl_data("fan-modules")
+        if fanmoduels_data:
+            for fan_module in fanmoduels_data:
+                uid = fan_module.get('durable-id', 'NA')
+                health = fan_module.get('health')
+                fan_module_resp = {
+                    "uid": uid,
+                    "fru": "true",
+                    "last_updated": int(time.time()),
+                    "health": {
+                        "status": health,
+                        "description": f"FAN is in {'good' if health=='OK' else 'bad'} health",
+                        "recommendation": fan_module.get('health-recommendation', 'NA'),
+                        "specifics": [self.get_fan_specfics(fan) for fan in fan_module['fan']],
+                    }
+                }
```

Comment:
Lets follow the recommendation in [comment](https://github.com/Seagate/cortx-monitor/pull/426#discussion_r653594550)
get_health_template()
set_health_data()

Reply:
agree, need PR to rebase & align to latest code

Reply:
Done.